lms/djangoapps/program_enrollments/api/reading.py

```diff
@@ -412,15 +415,19 @@ def get_users_by_external_keys_and_org_key(external_user_keys, org_key):
             saml_provider.get_social_auth_uid(external_user_key)
             for external_user_key in external_user_keys
         }
-        social_auths = UserSocialAuth.objects.filter(uid__in=social_auth_uids)
-        found_users_by_external_keys.update({
-            saml_provider.get_remote_id_from_social_auth(social_auth): social_auth.user
-            for social_auth in social_auths
-        })
+        if social_auth_uids:
+            # Filter should be case insensitive
+            query_filter = reduce(or_, [Q(uid__iexact=uid) for uid in social_auth_uids])
+            social_auths = UserSocialAuth.objects.filter(query_filter)
+            found_users_by_external_keys.update({
+                saml_provider.get_remote_id_from_social_auth(social_auth).lower(): social_auth.user
+                for social_auth in social_auths
+            })
 
     # Default all external keys to None, because external keys
     # without a User will not appear in `found_users_by_external_keys`.
-    users_by_external_keys = {key: None for key in external_user_keys}
+    # Return all keys as lowercase.
+    users_by_external_keys = {key.lower(): None for key in external_user_keys}
```

Comment:
this is somewhat suspect - if an external provider uses uppercase on purpose why would we change that?

Reply:
This is purely for search purposes - if I don't convert it here, it ends up returning a dict like `{'aBcDeF': None, 'AbCdEf': <User object>}`, where `aBcDeF` was the user supplied external_user_key. But it might make more sense to just convert it in the consuming function.